src/Control/Monad/Catch.hs

```diff
@@ -91,7 +91,12 @@ import Control.Monad.Trans.Reader (ReaderT(..), runReaderT)
 
 import GHC.Stack (HasCallStack, withFrozenCallStack)
 
+#if !defined(__MHS__)
 import Language.Haskell.TH.Syntax (Q)
+#else
+import Prelude hiding(foldr)
+import Data.Foldable
+#endif
```

Comment:
Minor style nit pick: do you mind inverting the order of the branches in this `#if` block? That is, write it like this instead:

```suggestion
#if defined(__MHS__)
import Prelude hiding(foldr)
import Data.Foldable
#else
import Language.Haskell.TH.Syntax (Q)
#endif
```